packages/as-types/assembly/amount.ts

```diff
@@ -28,62 +32,89 @@ export class Amount {
   /**
    * Adds two amounts and return results in a new one.
    *
-   * @param  a - Amount to add.
+   * @param  other - The other Amount to add.
+   *
+   * @returns
+   * - A Result containing the new Amount.
+   * - An error message if the currency is not the same between the two amounts.
+   * - An error message if the addition would overflow 'u64.MAX_VALUE'.
+   *
    */
-  add(a: Amount): Result<Amount> {
-    if (this.currency != a.currency) {
+  @operator('+')
+  add(other: Amount): Result<Amount> {
+    if (this.currency != other.currency) {
       return new Result(
         new Amount(),
         'the sum is impossible: the amounts have different currencies',
       );
     }
 
-    if (a.value > u64.MAX_VALUE - this.value) {
+    if (other.value > u64.MAX_VALUE - this.value) {
       // tests overflow
       return new Result(new Amount(), 'the sum is impossible: overflow');
     }
 
-    return new Result(new Amount(this.value + a.value, this.currency));
+    return new Result(new Amount(this.value + other.value, this.currency));
   }
 
   /**
-   * Substracts two amounts and return results in a new one.
+   * Subtracts two amounts and return results in a new one.
+   *
+   * @param  other - Amount to subtract.
    *
-   * @param  a - Amount to substract.
+   * @returns
+   * - A Result containing the new Amount.
+   * - An error message if the currency is not the same between the two amounts.
+   * - An error message if the subtraction would underflow .
    */
-  substract(a: Amount): Result<Amount> {
-    if (this.currency != a.currency) {
+  @operator('-')
+  subtract(other: Amount): Result<Amount> {
+    if (this.currency != other.currency) {
       return new Result(
         new Amount(),
         'the difference is impossible: the amounts have different currencies',
       );
     }
 
-    if (a.value > this.value) {
+    if (other.value > this.value) {
       // tests underflow
       return new Result(
         new Amount(),
         'the difference is impossible: underflow',
       );
     }
 
-    return new Result(new Amount(this.value - a.value, this.currency));
+    return new Result(new Amount(this.value - other.value, this.currency));
   }
 
   /**
    * Check if existent amount is lower than given one.
    *
-   * @param  a - Amount to check against.
+   * @param  other - Amount to check against.
+   *
+   * @returns true if the amount is lower than the given one.
+   *
+   * @throws if the currency is not the same between the two amounts.
```

Comment:
I don't like this throw. It will stop Sc execution I would prefer to return false; Wdyt?

Reply:
Hmm.. Idk, because its not an quality comparison but rather a quantity comparison.
Users might ignore the fact that the currencies are different and simply interpret that their result is "a is lessThan b" because it returned false and not take into account that currencies are different.
No ?

Reply:
Maybe we should use results in most operator cases and add an 'isSameCurrency' method ?

Reply:
We need to have a discussion on this. on monday ?